Add Windows installer partition support, minor improvements and related documentation

eugenesan opened the pull request.

eugenesan committed:
Add Windows installer partition support and related documentation

eugenesan committed:
Stop using subdirectories for most distributions

eugenesan committed:
Add boot device detection, support for multiple windows setup partitions and update windows icons

eugenesan committed:
Add info regarding Tails Persistent Storgage and readme cleanup

eugenesan committed:
Add option to skip partition order checking during install and minor tweak

eugenesan committed:
Fix missing hdnum typo

eugenesan committed:
Add more notes regarding Windows support

eugenesan committed:
Minor changes in install script and copy readme to boot drive

eugenesan committed:
Switch to non-strict partition order, add disk layout example and generic clenups

cshandley-uk commented:
Hi. I can see these big changes will be useful for some people (not me), but I am concerned that moving ISOs out of separate folders into a shared folder may make it harder to maintain GLIM & support new ISOs in the future (and it also makes it less clear which distro ISOs are supported).  I was thinking of going in the opposite direction, where GLIM would accept mostly any ISO filename, as long as it was in the correct folder (but I didn't want to make such a big change when it doesn't yet seem necessary & it might hamper the chances of my changes being merged upstream).  So at the moment I don't plan to merge your changes, sorry!  (But I will copy of a few of your small improvements/fixes.)

By the way, I have changed my GitHub's default branch from "DualPartitionSupport" to "master" (as GitHub had originally advised me), as I can see your fork ended-up with the same branch name (which I imagine can only cause problems for other forks).  I had planned to do this before mentioning it on the Ventoy thread, but you beat me to it.  I suggest you do what I did, after committing (or stashing) any new changes:
```
$ git checkout master
$ git merge DualPartitionSupport
$ git push
```
Before changing your default GitHub branch to "master".  You can then continue to make improvements on the "DualPartitionSupport" branch (or perhaps better create your own "MultiPartition" branch), and then occasionally merge those changes into "master" (once you've been tested better) so that other people (inc forks) can use them.  But I am not very familiar with the best way to handle GitHub, so maybe there is a better way...

BTW, I doubt your Windows icon is public domain or copyrighted by you, so I'm a bit doubtful it can be included with a GPLed program (see the end of the README.md).

eugenesan commented:
Thank you for your feedback.

My original idea behind using mostly single folder was to mimic Ventoy behavior which is what I and I suppose majority of potential users would expect. Not sure which of the 3 approaches (original, yours and mine) is the best...

I switched to and rebased against your master branch.
I also pulled a few commits from upstream PRs.

Regarding Windows logo it came from https://commons.wikimedia.org/wiki/File:Windows_10_Logo.svg# and if I understand correctly according to https://commons.wikimedia.org/wiki/File:Windows_10_Logo.svg#Licensing it is considered public domain.

BTW, why did you add `bash_` prefix to your clone? Is there a non-bash variant of the project?

cshandley-uk commented:
> BTW, why did you add bash_ prefix to your clone? Is there a non-bash variant of the project?

Everything I put on my GitHub page is organised by the language it's written in.  Since GitHub doesn't seem to support the concept of folders (that I could see), I've used prefixes instead.

> I switched to and rebased against your master branch.

Ah OK.  My plan is to keep my 'minimalist'(ish) changes in a separate branch (DualPartition), so that my pull request to the original project doesn't diverge too much.  Although the original project seems to have died, so maybe yours can take over...

> My original idea behind using mostly single folder was to mimic Ventoy behavior

I can see the attraction behind doing it like that, but I presume Ventoy has some additional magic to make random ISOs more likely to automagically work (beyond filename pattern matching).  My worry is that just using filename patterns will be very brittle, harder to debug, and probably a worse user experience when random ISOs fails to work (as users will need to RTFM - but at least you copied the readme!).
